scripts/simplify_network.py

```diff
@@ -423,7 +423,20 @@ def aggregate_to_substations(n, aggregation_strategies=dict(), buses_i=None):
         logger.info(
             "Aggregating buses that are no substations or have no valid offshore connection"
         )
-        buses_i = list(set(n.buses.index) - set(n.generators.bus) - set(n.loads.bus))
+        # isolated buses should be excluded from adjacency_matrix calculations
+        n.determine_network_topology()
+        # duplicated sub-networks mean that there is at least one interconnection between buses
+        i_islands = n.buses[
+            (~n.buses.duplicated(subset=["sub_network"], keep=False))
+            & (n.buses.carrier == "AC")
+        ].index
+
+        buses_i = list(
+            set(n.buses.index)
+            - set(i_islands)
+            - set(n.generators.bus)
+            - set(n.loads.bus)
+        )
```

Comment:
Did we have to modify this?

Reply:
Yes: we had to exclude isolated buses from treating with `aggregate_to_substations()` to resolves troubles which arise due to using inf values. Setting `inf` is used in `aggregate_to_substations()` to properly assign buses in a busmap. But in case of isolated buses, the whole string consists of `inf` and `idxmin()` picks an arbitrary bus